service/collector/app_operator.go

```diff
@@ -0,0 +1,56 @@
+package collector
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/kubernetes"
+)
+
+// AppOperatorConfig is this collector's configuration struct.
+type AppOperatorConfig struct {
+	K8sClient kubernetes.Interface
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+// AppOperator is the main struct for this collector.
+type AppOperator struct {
```

Comment:
Can't think about the right name but atm I just set it as `AppOperator`

Reply:
I think the name should be `AppResource` to match what we do in chart-operator. https://github.com/giantswarm/chart-operator/blob/master/service/collector/chart_resource.go

I think this will be the first metric but its fine for it to be just the empty structure for now.